src/app/server/Server.cpp

```diff
@@ -238,6 +232,40 @@ CHIP_ERROR Server::Init(AppDelegate * delegate, uint16_t secureServicePort, uint
 
     err = mCASESessionManager.Init();
 
+    // This code is necessary to restart listening to existing groups after a reboot
+    // Each manufacturer needs to validate that they can rejoin groups by placing this code at the appropriate location for them
+    //
+    // This is disabled for thread device because the same code is already present for thread devices in
```

Comment:
Sounds like this code should be factored out into a helper function of some sort (on Server, perhaps?).

Especially because the code here and in GenericThreadStackManagerImpl_OpenThread_LwIP.cpp doesn't have the same behavior: on failure to join some group this code stops trying completely while the OpenThread code just moves on to the next fabric.

Please file a followup for this?

Reply:
Issue was created
https://github.com/project-chip/connectedhomeip/issues/14254